apps/web/src/app/(private)/evaluations/[evaluationUuid]/editor/_components/EvaluationEditor/Editor/index.tsx

```diff
@@ -61,7 +62,7 @@ export default function EvaluationEditor({
   useEffect(() => {
     readMetadata({
       prompt: value,
-      withParameters: Object.keys(PARAMETERS_FROM_LOG),
+      // withParameters: ['messages', 'response'], // TODO: uncomment this line once I understand why it breaks
```

Comment:
cc @csansoon you probably can provide some light here

Reply:
Unlike with documents, evaluations do not have the ability to create new variables just by referencing them in the code. Instead, it has a bunch of predefined variables, and users can only use those. This means that, to "readMetadata", the compiler must know which are these variables names so that it knows if it would fail when you reference a parameters. If it exists in that list nothing happens, but if it doesn't, it should fail with a "variable not declared".

Reply:
Unlike with documents, evaluations do not have the ability to create new variables just by referencing them in the code. Instead, it has a bunch of predefined variables, and users can only use those. This means that, to "readMetadata", the compiler must know which are these variables names so that it knows if it would fail when you reference a parameters. If it exists in that list nothing happens, but if it doesn't, it should fail with a "variable not declared".

Reply:
yes that's what i expected but the compiler fails with a "messages variable not declared" in this document, why?

![image](https://github.com/user-attachments/assets/5a7c0629-9575-469e-b69b-2063d883201c)

Reply:
well, if `withParameters` contains `messages`, that's a bug in the compiler then 😅

Reply:
talked offline, will fix it in the next pr